massa-execution-worker/src/speculative_roll_state.rs

```diff
@@ -148,6 +149,63 @@ impl SpeculativeRollState {
         Ok(())
     }
 
+    /// Try to slash `roll_count` rolls from the given address. If not enough roll, slash
+    /// the available amount and return the value.
+    ///
+    /// # Arguments
+    /// * `addr`: address to slash the rolls from
+    /// * `roll_count`: number of rolls to slash
+    pub fn try_slash_rolls(
+        &mut self,
+        addr: &Address,
+        roll_count: u64,
+    ) -> Result<u64, ExecutionError> {
+        // fetch the roll count
+        let owned_count = self.get_rolls(addr);
+        let roll_to_slash = min(roll_count, owned_count);
+
+        // remove the rolls
+        let current_rolls = self
+            .added_changes
+            .roll_changes
+            .get_mut(addr)
+            .ok_or_else(|| {
+                ExecutionError::RuntimeError(format!("Cannot get roll changes for addr: {}", addr))
```

Comment:
if the address does not exist anymore, let's not end into a runtime error, but simply don't slash anything

Reply:
Removed